### PR TITLE
Fixing cloud-data deserialization crash

### PR DESCRIPTION
**Pull request by robbiehanson**
The problem is that an exception thrown in Kotlin isn't caught in Swift:

```swift
// Swift code
do {
  let wrapper = CloudData.companion.cborDeserialize(blob: blob)
  // ...
} catch {
  // Kotlin exception will NOT be caught here :(
}
```

Instead the app will simply crash. So the solution, for now, is to have the Kotlin function use a try/catch, and change the return type to an optional:

```kotlin
fun CloudData.Companion.cborDeserialize(
  blob: ByteArray
): Pair<CloudData?, WalletPayment?> {
  val wrapper = try {
    Cbor { ignoreUnknownKeys = true}.decodeFromByteArray<CloudData>(blob)
  } catch (_: Throwable) {
    return Pair(null, null)
  }
  val payment = try {
    wrapper.unwrap()
  } catch (_: Throwable) {
    null
  }
  return Pair(wrapper, payment)
}

```

### CODE REVIEW

**Comment by dpad85** [comment]
Have you tried annotating `cborDeserialize` with `@Throws`? It should automatically convert any annotated exception to a `NSError`.

See https://kotlinlang.org/docs/native-objc-interop.html#errors-and-exceptions

**Comment by robbiehanson** [comment]
Good idea. I think that actually works better. So I've adopted that approach in the most recent commit.

**Comment by robbiehanson** [comment]
Force-pushed because I forgot to sign the first 3 commits

File: phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/db/cloud/CloudAsset.kt [hunk 1]
```diff
@@ -111,9 +107,7 @@ fun WalletPaymentMetadataRow.cloudSerialize(): ByteArray {
 @OptIn(ExperimentalSerializationApi::class)
 fun CloudAsset.Companion.cloudDeserialize(blob: ByteArray): WalletPaymentMetadataRow? {
     val wrapper: CloudAsset = try {
-        Cbor {
-            ignoreUnknownKeys = true
-        }.decodeFromByteArray(blob)
+        Cbor { ignoreUnknownKeys = true }.decodeFromByteArray(blob)
```
[the remark this answers is not in the document]
Comment (robbiehanson) [reply phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/db/cloud/CloudAsset.kt]: Done in [dff8edf](https://github.com/ACINQ/phoenix/pull/332/commits/dff8edfbf7e71fbc9d365fbafa182104619bc071)

**Review by dpad85 — APPROVED** [review]

**Pull request merged**